src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -992,6 +993,233 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_JoinerStart(voi
+        }
+    }
+
+    // Verify is there an empty place for new service.
+    VerifyOrExit(srpService, error = OT_ERROR_NO_BUFS);
+
+    memset(srpService, 0, sizeof(otSrpClientService));
+    otSrpClientSetLeaseInterval(mOTInst, aLeaseInterval);
+    otSrpClientSetKeyLeaseInterval(mOTInst, aKeyLeaseInterval);
+
+    srpService->mInstanceName = aInstanceName;
+    srpService->mName         = aName;
+    srpService->mPort         = aPort;
+
+    error = otSrpClientAddService(mOTInst, srpService);
+
+exit:
+    return MapOpenThreadError(error);
+}
+
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_RemoveSrpService(const char * aInstanceName, const char * aName)
+{
+    otError error                   = OT_ERROR_NONE;
+    otSrpClientService * srpService = nullptr;
+
+    VerifyOrExit(aInstanceName, error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(aName, error = OT_ERROR_INVALID_ARGS);
+
+    // Check if service to remove exists.
+    for (otSrpClientService & service : mSrpServices)
+    {
+        if ((strcmp(service.mInstanceName, aInstanceName) == 0) && (strcmp(service.mName, aName) == 0))
+        {
+            srpService = &service;
+            break;
+        }
+    }
+
+    VerifyOrExit(srpService, error = OT_ERROR_NOT_FOUND);
+
+    error = otSrpClientRemoveService(mOTInst, srpService);
+
+exit:
+    return MapOpenThreadError(error);
+}
+
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_StartSrpClient(const char * aHostName)
+{
+    otError error = OT_ERROR_NONE;
+    otIp6Address clientIpAddress;
+    otServiceConfig serverServiceConfig;
+
+    VerifyOrExit(aHostName, error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(!otSrpClientIsRunning(mOTInst), error = OT_ERROR_ALREADY);
+
+    memcpy(&clientIpAddress, otThreadGetMeshLocalEid(mOTInst), sizeof(clientIpAddress));
+
+    error = otSrpClientSetHostName(mOTInst, aHostName);
+    VerifyOrExit(error == OT_ERROR_NONE, );
+
+    error = otSrpClientSetHostAddresses(mOTInst, &clientIpAddress, 1);
+    VerifyOrExit(error == OT_ERROR_NONE, );
+
+    VerifyOrExit(FindSrpServerService(serverServiceConfig), error = OT_ERROR_NOT_FOUND);
+
+    // Convert server service config data to get from it server IP address and port number.
+    otSockAddr serverSocketAddress;
+    serverSocketAddress.mAddress =
+        GenericThreadStackManagerImpl_OpenThread::ConvertSrpServerRloc16ToIpAddress(serverServiceConfig.mServerConfig.mRloc16);
+    serverSocketAddress.mPort =
+        GenericThreadStackManagerImpl_OpenThread::ConvertSrpServerDataToPort(serverServiceConfig.mServerConfig.mServerData);
+
+    error = otSrpClientStart(mOTInst, &serverSocketAddress);
+
+exit:
+    return MapOpenThreadError(error);
+}
+
+template <class ImplClass>
+bool GenericThreadStackManagerImpl_OpenThread<ImplClass>::FindSrpServerService(otServiceConfig & aServiceConfig)
+{
+    bool serviceFound = false;
+    Impl()->LockThreadStack();
```

Comment:
Shouldn't we rather guard the public methods like `AddSrpService` `RemoveSrpService`... ?